js/sparkswap.js

```diff
@@ -271,7 +271,46 @@ module.exports = class sparkswap extends Exchange {
     }
 
     async fetchOrder (id, symbol = undefined, params = {}) {
-        return this.privateGetV1OrderId ({ id });
+        const order = await this.privateGetV1OrdersId ({ id });
+        let status = 'open';
```

Comment:
I don't agree with setting defaults like this. Why is open the default? Why not just look at what the order.status is and figure it out from there. Here we are making an assumption that if the order status is not cancelled or completed that the status should be open. I don't think that is a fair assumption to make considering it could be failed, or the data could just be messed up. 

Setting these defaults feels confusing to me.

Reply:
@martineehrlich Totally agree. I missed the case for `failed` as well.